### ISSUE TITLE
CrossGen produces spurious "Error" logging on what it later reports as successful compilation

### ISSUE THREAD

**Issue by MattGal** [dotnet/coreclr#19704]
From this log: https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_apis/build/builds/1980843/logs/8

```
2018-08-27T11:42:26.3776644Z          Microsoft (R) CoreCLR Native Image Generator - Version 4.5.30319.0
2018-08-27T11:42:26.3822938Z          Copyright (c) Microsoft Corporation.  All rights reserved.
2018-08-27T11:42:26.3869273Z          
2018-08-27T11:42:26.3912527Z          Error 2147500037 (Exception from HRESULT: 0x80004005 (E_FAIL)) while compiling method BuildServerConnection.CreateMonitorDisconnectTask
2018-08-27T11:42:26.3958431Z          Error 2147500037 (Exception from HRESULT: 0x80004005 (E_FAIL)) while compiling method BuildRequest.WriteAsync
2018-08-27T11:42:26.4003208Z          Error 2147500037 (Exception from HRESULT: 0x80004005 (E_FAIL)) while compiling method BuildResponse.WriteAsync
2018-08-27T11:42:26.4048083Z          Error 2147500037 (Exception from HRESULT: 0x80004005 (E_FAIL)) while compiling method BuildProtocolConstants.ReadAllAsync
2018-08-27T11:42:26.4100353Z          Native image /data/agent/_work/_temp/tmpRZTXCg.tmp generated successfully.
```
(This was from compiling csc.dll)

Some thoughts here:
- If four errors have been written out to the console, succeeding with no special "-force' or other args like this is very confusing
- MSBuild and other systems tend to promote std. out with "error" in the string to actual errors
- E_FAIL is not super useful to anyone reading this log.

@eerhardt FYI

**Comment by zamont** [comment]
@fadimounir bringing to your attention. I think you may have already resolved this with some of your recent changes.

**Comment by fadimounir** [comment]
This issue should be already fixed via #21987. If this still repros, please reopen this issue and provide me with repro instructions (where to get dlls from, crossgen command line, etc...)
